### PR TITLE
Adds beetlejuice component.

### PR DESCRIPTION
**Pull request by AnturK**
Say the name x times to make it appear.

### CODE REVIEW

**Comment by AnturK** [comment]
Seems NAMEOF has some issues here.

**Comment by GuyonBroadway** [comment]
oranges
oranges
oranges

**Comment by MrPerson** [comment]
Don't be silly, everyone knows oranges isn't real.

**Comment by nicbn** [comment]
ocean man ocean man ocean man

**Comment by ExcessiveUseOfCobblestone** [comment]
Legend has it if you look in the dorms mirror and say 4dplanner thrice, your job immediately gets nerfed.

**Comment by davethwave** [comment]
Say woody got wood enough times and you get a free smite?

**Comment by dr-bee** [comment]
> 
> 
> Say woody got wood enough times and you get a free smite?

Fully automated BSA for woody spammers.

File: code/datums/components/beetlejuice.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/datum/component/beetlejuice/proc/say_react(datum/source, mob/speaker,message)
+	if(!speaker || !message || !active)
+		return
+	var/found = findtext(message,keyword)
+	if(found)
+		var/occurences = 0
+		while(found > 0)
+			occurences++
+			found = findtext(message,keyword,found + length(keyword) + 1)
```
Comment (optimumtact) [inline code/datums/components/beetlejuice.dm]: regex?

**Comment by GuyonBroadway** [comment]
It seems this component has performance issues, it took six days between me saying it and the orange man turning up.

File: code/datums/components/beetlejuice.dm (hunk 1 above)
Comment (AnturK) [reply code/datums/components/beetlejuice.dm]: I'd have to hookup varedits to recompile but alright

**Pull request merged**